tests/unit/multi_table/test_hma.py

```diff
@@ -85,6 +85,42 @@ def test__get_extension(self):
 
         pd.testing.assert_frame_equal(result, expected)
 
+    def test_warning_message_too_many_cols(self, capsys):
+        """Test that a warning appears if there are more than 1000 expected columns"""
+
+        metadata = get_multi_table_metadata()
+
+        key_phrases = [
+            r'PerformanceAlert:',
+            r'large number of columns.',
+            r'contact us at info@sdv.dev for enterprise solutions.'
+        ]
+
+        with patch.object(HMASynthesizer,
+                          '_estimate_num_columns',
+                          return_value={'nesreca': 2000}):
+            HMASynthesizer(metadata)
+
+        captured = capsys.readouterr()
+
+        # Assert
+        for pattern in key_phrases:
+            match = re.search(pattern, captured.out + captured.err)
+            assert match is not None
+
+        small_metadata = get_multi_table_metadata()
+        with patch.object(HMASynthesizer,
```

Comment:
Usually we use the decorator `@patch` but as you prefer here

Reply:
I used the patch.object instead of the annotation because I need to adjust the return value to test both branches of the print statement (low columns vs high columns).